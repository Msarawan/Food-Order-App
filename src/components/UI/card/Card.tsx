import * as React from 'react';
import classes from './Card.module.css';


interface ICardProps{
  children:any
}

const Card: React.FC<ICardProps> = ({children}) => {
  return <div className={classes.card}>{children}</div>
};

export default Card;