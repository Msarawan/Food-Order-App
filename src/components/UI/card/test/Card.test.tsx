//import 'jsdom-global/register';
import Card from "../Card";
//import AvailableMeals from "../../../Meals/availableMeal/AvailableMeals";
import React from "react";
//import Adapter from 'enzyme-adapter-react-16';
import { mount} from "enzyme";

describe("Food Order app testing", () => {
  let wrapper;
  test("render the childrens of the props", () => {
    const mealsList = 
      [
        {
          id: 'm1',
          name: 'Aloo Kulcha',
          description: 'Whole wheat flour, potatoes,garam masala',
          price: 22.99,
        },
        {
          id: 'm2',
          name: 'Veg Fried Rice',
          description: 'Matar, basmati rice, cabbage, soy sauce, beans',
          price: 16.5,
        },
        {
          id: 'm3',
          name: 'Barbecue Burger',
          description: 'American, raw, meaty',
          price: 12.99,
        },
        {
          id: 'm4',
          name: 'Chicken Sizzler',
          description: 'Sizzler, skinless chicken breasts, cherry tomatoes',
          price: 18.99,
        },
      ];

      
      wrapper = mount(<Card item ={mealsList}/>);
      expect(wrapper).toMatchSnapshot();
      expect(mealsList).toHaveLength(4);
     console.log(wrapper.find('Card'))
  });
})


