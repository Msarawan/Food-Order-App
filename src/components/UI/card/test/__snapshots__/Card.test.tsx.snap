// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Order app testing render the childrens of the props 1`] = `
<Card>
  <div
    className="card"
  />
</Card>
`;
